packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -245,20 +271,30 @@ class UpgradeCommandRunner {
       final String errorString = e.toString();
       if (errorString.contains('fatal: HEAD does not point to a branch')) {
         throwToolExit(
-          'You are not currently on a release branch. Use git to '
-          'check out an official branch (\'stable\', \'beta\', \'dev\', or \'master\') '
-          'and retry, for example:\n'
+          'Unable to upgrade Flutter: HEAD does not point to a branch(Are you '
```

Comment:
I'm not sure this error message was accurately being displayed when the user is not on a release branch (branches other than `kOfficialChannels`). I changed this to only warn about detached HEAD states.

Reply:
If the user is on a detached HEAD, they would also not be on a release branch right? Even if this message is more specific, I doubt most casual git users understand what a detached HEAD state is, when what is relevant to their use of Flutter is that they are not on an official channel. What do you think?

Reply:
Also, while we're at it, we can advise the user to use `flutter channel stable`, rather than doing a git checkout directly.

Reply:
Done.